perly.y

```diff
@@ -987,8 +984,12 @@ sigsubbody:	remember optsubsignature PERLY_BRACE_OPEN stmtseq PERLY_BRACE_CLOSE
 /* Ordinary expressions; logical combinations */
 expr	:	expr[lhs] ANDOP expr[rhs]
 			{ $$ = newLOGOP(OP_AND, 0, $lhs, $rhs); }
+	|	expr[lhs] PLUGLOWANDOP[op] expr[rhs]
+			{ $$ = build_infix_plugin($lhs, $rhs, $op); }
 	|	expr[lhs] OROP[operator] expr[rhs]
 			{ $$ = newLOGOP($operator, 0, $lhs, $rhs); }
+	|	expr[lhs] PLUGLOWOROP[op] expr[rhs]
```

Comment:
What is `OROP[operator]` doing above here that you aren't and does it matter?

Reply:
The newly-added rules are all the ones consuming a `PLUG*` token, which means it's a plugin infix operator handled by `PL_infix_plugin`. The code blocks all have to run `build_infix_plugin()` to invoke the code properly. As compared the non-`PLUG*` ones which just build internal native optrees directly by calling e.g. `newLOGOP` in this case.

Reply:
Plus: Yes it's annoying that this is basically the same exact rule copied nine times with nine different token types. I did try squashing them all together with a single rule inbetween, but that causes a huge amount of grammar conflict, because that loses operator precedence information. :(